packages/react/src/QuestionnaireForm/QuestionnaireForm.tsx

```diff
@@ -61,22 +66,38 @@ export function QuestionnaireForm(props: QuestionnaireFormProps): JSX.Element |
     }
   }, [response]);
 
-  const setItems = useCallback((newResponseItems: QuestionnaireResponseItem | QuestionnaireResponseItem[]): void => {
-    setResponse((prevResponse) => {
-      const currentItems = prevResponse?.item ?? [];
-      const mergedItems = mergeItems(
-        currentItems,
-        Array.isArray(newResponseItems) ? newResponseItems : [newResponseItems]
-      );
-
-      const newResponse: QuestionnaireResponse = {
-        resourceType: 'QuestionnaireResponse',
-        status: 'in-progress',
-        item: mergedItems,
-      };
-      return newResponse;
-    });
-  }, []);
+  const setItems = useCallback(
+    (newResponseItems: QuestionnaireResponseItem | QuestionnaireResponseItem[]): void => {
+      setResponse((prevResponse) => {
+        const currentItems = prevResponse?.item ?? [];
+        const mergedItems = mergeItems(
+          currentItems,
+          Array.isArray(newResponseItems) ? newResponseItems : [newResponseItems]
+        );
+
+        const tempResponse: QuestionnaireResponse = {
+          resourceType: 'QuestionnaireResponse',
+          status: 'in-progress',
+          item: mergedItems,
+        };
+
+        const updatedItems = evaluateCalculatedExpressions(questionnaire?.item ?? [], tempResponse);
+        const mergedItemsWithUpdates = mergedItems.map((mergedItem) => {
+          const updatedItem = updatedItems.find((updated) => updated.linkId === mergedItem.linkId);
+          return updatedItem || mergedItem;
```

Comment:
I think there was some discussion the other day about how to resolve conflicts when a calculated response on the same item that the user answers. It'd be good to add some comments above this code explaining that we decided the calculated response always wins. If there's any excerpts you could add from the FHIR spec to support that decision, that would be helpful.